src/algorithm/compute-all-terms.hpp

```diff
@@ -233,6 +233,9 @@ namespace pinocchio
       Pass2::run(model.joints[i],data.joints[i],
                  typename Pass2::ArgsType(model,data));
     }
+
+    //Add the armature value. By default, this is a zero vector.
```

Comment:
You should add an assert here, or in the checking of the algo, to check if the armature is indeed positive.

Reply:
Done.